src/components/NewCommentForm.tsx

```diff
@@ -1,13 +1,17 @@
 import classNames from 'classnames';
 import React, { useState } from 'react';
-import { CommentData } from '../types/Comment';
+import { Comment } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { addCommentAPI } from '../features/CommentsAPI/CommentsAPI';
 
-type Props = {
-  onSubmit: (data: CommentData) => Promise<void>;
-};
-
-export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
+export const NewCommentForm: React.FC = () => {
   const [submitting, setSubmitting] = useState(false);
+  const dispatch = useAppDispatch();
+  const selectedPost = useAppSelector(state => state.currentPost.post);
+
+  const addComment = async (commentData: Omit<Comment, 'id'>) => {
+    await dispatch(addCommentAPI(commentData as Comment));
```

Comment:
The `addCommentAPI` function is being dispatched with `commentData` cast as `Comment`. This could potentially lead to runtime errors if `commentData` does not have all the properties of `Comment`. Ensure that `commentData` is correctly structured before dispatching.